File: funnyvote/app/src/main/java/com/heaton/funnyvote/ui/about/AboutPresenter.java
package com.heaton.funnyvote.ui.about;

public class AboutPresenter implements AboutContract.Presenter {

    private AboutContract.View view;

    public AboutPresenter(AboutContract.View view) {
        this.view = view;
        this.view.setPresenter(this);
    }

    @Override
    public void IntentToIntroduction() {
        view.showIntroduction();
    }

    @Override
    public void IntentToAuthorInfo() {
        view.showAuthorInfo();
    }

    @Override
    public void IntentToLicence() {
        view.showLicence();
    }

    @Override
    public void IntentToProblem() {
        view.showProblem();
    }

    @Override
    public void IntentToAppStore() {
        view.showAppStore();
    }

    @Override
    public void IntentToAbout() {
        view.showAbout();
    }

    @Override
    public void IntentToShareApp() {
        view.showShareApp();
    }

    @Override
    public void subscribe() {

    }

    @Override
    public void unsubscribe() {

    }
}

File: funnyvote/app/src/main/java/com/heaton/funnyvote/ui/about/AboutPresenter.kt
package com.heaton.funnyvote.ui.about

class AboutPresenter(private val view: AboutContract.View) : AboutContract.Presenter {

    init {
        this.view.setPresenter(this)
    }

    override fun IntentToIntroduction() {
        view.showIntroduction()
    }

    override fun IntentToAuthorInfo() {
        view.showAuthorInfo()
    }

    override fun IntentToLicence() {
        view.showLicence()
    }

    override fun IntentToProblem() {
        view.showProblem()
    }

    override fun IntentToAppStore() {
        view.showAppStore()
    }

    override fun IntentToAbout() {
        view.showAbout()
    }

    override fun IntentToShareApp() {
        view.showShareApp()
    }

    override fun start() {

    }
}
